.github/workflows/deploy.yml

```diff
@@ -0,0 +1,96 @@
+name: deploy
+
+on:
+  push:
+    branches:
+    - '**'
+    tags:
+    - 'v*'
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  goreleaser:
+    name: goreleaser
+    runs-on: ubuntu-18.04
+    environment: deploy
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: install
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.15.5
+
+    - name: release
+      run: |
+        if [[ $GITHUB_REF != refs/tags/* ]]; then
+          export DRY_RUN=1
+        fi
+        .ci/gpg/create-keyring.sh
+        make release
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GPG_PASSWORD: ${{ secrets.GPG_PASSWORD }}
+
+  images:
+    name: images
+    runs-on: ubuntu-18.04
+    environment: deploy
+    strategy:
+      matrix:
+        id: ["operator-sdk", "ansible-operator", "helm-operator", "scorecard-test", "scorecard-test-kuttl"]
+    steps:
+
+    - name: set up qemu
+      uses: docker/setup-qemu-action@v1
+
+    - name: set up buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: quay.io login
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.QUAY_USERNAME }}
+        password: ${{ secrets.QUAY_PASSWORD }}
+        registry: quay.io
+
+    - name: create tags
+      id: tags
+      run: |
+        IMG=quay.io/operator-framework/${{ matrix.id }}
+        if [[ $GITHUB_REF == refs/tags/* ]]; then
+          TAG=${GITHUB_REF#refs/tags/}
+          MAJOR_MINOR=${TAG%.*}
+          echo ::set-output name=tags::${IMG}:${TAG},${IMG}:${MAJOR_MINOR}
+
+        elif [[ $GITHUB_REF == refs/heads/* ]]; then
+          TAG=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+          if [ "${{ github.event.repository.default_branch }}" = "$TAG" ]; then
+            TAG=edge
+          fi
```

Comment:
Consider removing this if you want to stick with branch based tag name, e.g. `quay.io/operator-framework/operator-sdk:master`